Vostok.ClusterClient.Core/Modules/LoggingModule.cs

```diff
@@ -79,6 +88,93 @@ private void LogFailedResult(IRequestContext context, ClusterResult result)
+        {
+            const string template = "'{Request}' to '{TargetService}'. Code = {ResponseCode:D}. Time = {ElapsedTime}.";
+            var properties = new
+            {
+                Request = context.Request.ToString(false, false),
+                TargetService = targetService ?? "somewhere",
+                ResponseCode = result.Response.Code,
+                ElapsedTime = context.Budget.Elapsed.ToPrettyString(),
+                ElapsedTimeMs = context.Budget.Elapsed.TotalMilliseconds
+            };
+
+            switch (result.Status)
+            {
+                case ClusterResultStatus.Success:
+                    context.Log.Info(template, properties);
+                    break;
+                case ClusterResultStatus.Canceled:
+                    context.Log.Warn(template, properties);
+                    break;
+                default:
+                    context.Log.Error(template, properties);
+                    break;
+            }
+        }
+
+        private void LogVerboseMessage(IRequestContext context, ClusterResult result)
+        {
+            const string template = "'{Request}' to '{TargetService}', Timeout = {Timeout}. {Status} in {ElapsedTime}, Code = {ResponseCode:D}. Replicas result = {ReplicasResult}";
+            var properties = new
+            {
+                Request = context.Request.ToString(false, false),
+                TargetService = targetService ?? "somewhere",
+                Timeout = context.Budget.Total.ToPrettyString(),
+                TimeoutMs = context.Budget.Total.TotalMilliseconds,
+                result.Status,
+                ElapsedTime = context.Budget.Elapsed.ToPrettyString(),
+                ElapsedTimeMs = context.Budget.Elapsed.TotalMilliseconds,
+                ResponseCode = result.Response.Code,
+                ReplicasResult = ConvertResultsToProperties(result.ReplicaResults)
+            };
+
+            switch (result.Status)
+            {
+                case ClusterResultStatus.Success:
+                    context.Log.Info(template, properties);
+                    break;
+                case ClusterResultStatus.Canceled:
+                    context.Log.Warn(template, properties);
+                    break;
+                default:
+                    context.Log.Error(template, properties);
+                    break;
+            }
+        }
+
+        private object ConvertResultsToProperties(IList<ReplicaResult> replicaResults)
+        {
+            var properties = new object[replicaResults.Count];
+            for (var i = 0; i < replicaResults.Count; i++)
+            {
+                var res = replicaResults[i];
+                properties[i] = new
+                {
+                    res.Replica,
+                    ResponseCode = (int)res.Response.Code,
+                    res.Verdict,
+                    ElapsedTime = res.Time
```

Comment:
```suggestion
                    ElapsedTime = res.Time.ToPrettyString()
```
И ещё ElapsedTimeMs?

Reply:
Не, ElapsedTimeMs не нужен. Он же запринтится в текст, зачем нам два варианта написания времени. Это свойство же не для структурированно гологирования, а для того, чтобы в текст напечатать.